apps/module-presale/contracts/Presale.sol

```diff
@@ -0,0 +1,336 @@
+    string private constant ERROR_INVALID_TIME_PERIOD            = "PRESALE_INVALID_TIME_PERIOD";
+    string private constant ERROR_INVALID_DAI_FUNDING_GOAL       = "PRESALE_INVALID_DAI_FUNDING_GOAL";
+    string private constant ERROR_INVALID_PERCENT_VALUE          = "PRESALE_INVALID_PERCENT_VALUE";
+    string private constant ERROR_INVALID_POOL                   = "PRESALE_INVALID_POOL";
+    string private constant ERROR_INVALID_BENEFICIARY_ADDRESS    = "PRESALE_INVALID_BENEFICIARY_ADDRESS";
+    string private constant ERROR_EXCEEDS_FUNDING_GOAL           = "PRESALE_EXCEEDS_FUNDING_GOAL";
+
+    /*
+     * Roles
+     */
+
+    bytes32 public constant START_ROLE = keccak256("START_ROLE");
+    bytes32 public constant BUY_ROLE   = keccak256("BUY_ROLE");
+
+    /*
+     * Properties
+     */
+
+    // Token that will be accepted as payment for
+    // purchasing project tokens.
+    ERC20 public daiToken;
+
+    // Token that is being offered for purchase in the sale.
+    MiniMeToken public projectToken;
+    TokenManager public projectTokenManager;
+
+    // Funding goal and total dai raised in the sale.
+    // Note: no further purchases will be allowed after daiFundingGoal is reached.
+    uint256 public daiFundingGoal;
+    uint256 public totalDaiRaised;
+
+    // Percentage of the total supply of project tokens that will be offered to contributors,
+    // in this pre-sale and in further fundraising stages.
+    uint256 public percentSupplyOffered; // Represented in PPM, see below
+
+    // Once the sale is closed, totalDaiRaised is split according to
+    // percentFundingForBeneficiary, between beneficiaryAddress and fundraisingPool.
+    address public fundraisingPool;
+    address public beneficiaryAddress;
+    uint256 public percentFundingForBeneficiary; // Represented in PPM, see below
+
+    // Date when the sale is started and its state is Funding.
+    uint64 public startDate;
+
+    // Vesting parameters.
+    // Note: startDate also represents the starting date for all vested project tokens.
+    uint64 public vestingCliffPeriod;
+    uint64 public vestingCompletePeriod;
+
+    // Period after startDate, in which the sale is Funding and accepts contributions.
+    // If the daiFundingGoal is not reached within it, the sale cannot be Closed
+    // and the state switches to Refunding, allowing dai refunds.
+    uint64 public fundingPeriod;
+
+    // Number of project tokens that will be sold for each dai.
+    // Calculated after initialization from the values CONNECTOR_WEIGHT_PPM and percentSupplyOffered.
+    uint256 public daiToProjectTokenExchangeRate;
+
+    // Percentages are represented in the PPM range (Parts per Million) [0, 1000000]
+    // 25% => 0.25 * 1e6
+    // 50% => 0.50 * 1e6
+    uint256 public constant PPM = 1000000;
+
+    // Used to calculate daiToProjectTokenExchangeRate.
+    uint256 public constant CONNECTOR_WEIGHT_PPM = 100000; // 10%
```

Comment:
It may make sense to move these constants to the start of the file, after the roles.